src/core/compact_object.h

```diff
@@ -374,8 +377,12 @@ class CompactObj {
   } __attribute__((packed));
 
   struct JsonWrapper {
-    JsonType* json_ptr = nullptr;
-    size_t unneeded = 0;
+    union {
+      JsonType* json_ptr;
+      uint8_t* flat_ptr;
+    };
+    uint32_t json_len = 0;
+    uint8_t encoding = 0;
   } __attribute__((packed));
```

Comment:
What about ARM?

Reply:
what about it?

Reply:
This attribute changes alignment, but in this case it looks do nothing

Reply:
ARM doesn't like incorrect allignment

Reply:
move the attribute to more relevant place (line 399) and added a comment.

Reply:
We had this code since the inception and it worked fine on arm.

Reply:
because it does nothing in this case)

Reply:
The problem can be if you reorder elements